### DISCUSSION TITLE
Is there a way in JS to give a class name and get a result of the computed rules?

### DISCUSSION THREAD

**Discussion by mririgoyen** [tailwindlabs/tailwindcss#12241] [Help] [answered]
This is a very niche use case.

We have a Storybook that outlines typographical elements. We want to be able to display the pixel values for the font size, line height, etc. in plaintext. Instead of hard coding those values in, I'd really like to be able to just pull the computed values from Tailwind based on the class name used for each example. That way, the classnames are the source of truth and we don't have to maintain a separate, hard-coded list of values.

I was hoping there was some utility built in to Tailwind to do this, but I'm not finding anything publicly available. Something along the lines of this pseudocode:

```js
import { translateClassName } from 'tailwindcss';

const rules = translateClassName('text-xl/8');

// Output:
// {
//   'font-size': '1.25rem',
//   'line-height': '2rem',
// }
```

**Comment by wongjn** [comment, marked as answer]
You could try using the PostCSS API to programmatically run Tailwind and then use something like [`postcss-js`](https://github.com/postcss/postcss-js) to parse it back to an object:

```js
const postcss = require('postcss');
const tailwindcss = require('tailwindcss');
const postcssJs = require('postcss-js');
const baseConfig = require('./tailwind.config');

const config = {
  presets: [baseConfig],
  content: [{ raw: 'text-xl/8' }],
};
postcss([tailwindcss(config)])
  .process('@tailwind utilities')
  .then((css) => {
    const root = postcss.parse(css);
    const result = postcssJs.objectify(root);
    // {
    //   '.text-xl/8': {
    //     'font-size': '1.25rem',
    //     'line-height': '2rem',
    //   },
    // }
  });
```
Otherwise, there is no (at least public) way of intercepting/resolving CSS rules in Tailwind by itself.

**Comment by mririgoyen** [comment]
Thanks @wongjn, this looks promising. However, it isn't showing me any utilities that are registered via the `addUtilities` function via a plugin. Do you know how I might target that in the `process` step?

**Reply by mririgoyen** [reply]
I think I got it. I needed to put the class name I was looking for in the `content` section, not the `process` call. Thank you!

**Comment by sadnansami** [comment]
https://github.com/Arthie/xwind?tab=readme-ov-file

This should get the job done. There's different modes
